src/include/gurt/common.h

```diff
@@ -204,13 +204,65 @@ extern "C" {
 #define D_ALLOC_ARRAY(ptr, count) D_ALLOC_CORE(ptr, sizeof(*ptr), count)
 #define D_FREE_PTR(ptr)		D_FREE(ptr)
 
-#define D_GOTO(label, rc)			\
-	do {					\
-		__typeof__(rc) __rc = (rc);		\
-		(void)(__rc);			\
-		goto label;			\
+/* _Generic() is part of the c11 standard but isn't yet supported by
+ * gcc until version 4.9 so check for that but enable it in all other
+ * cases.
+ */
+
+#ifdef __clang__
+#define VERBOSE_GOTO 1
+#else
+#if defined  __GNUC__ && __GNUC__ > 4 || \
+	(__GNUC__ == 4 && (__GNUC_MINOR__ < 9))
+#define VERBOSE_GOTO 0
+#else
+#define VERBOSE_GOTO 1
+#endif
+#endif
+
+#if VERBOSE_GOTO
+
+#define maybe_derrcode(x) _Generic((x), int: true, default: false)
+
+#define DG_FMT(x) _Generic(__rc,					\
+				int: "%d",				\
+				unsigned int: "%u",			\
+				unsigned long: "%lu",			\
+				bool: "%d",				\
+				int64_t: "%ld", default: "%p")
```

Comment:
(style)  spaces required around that ':' (ctx:VxW)
(style)  spaces required around that ':' (ctx:VxW)
(style)  labels should not be indented